lib/util/PathSet.js

```diff
@@ -0,0 +1,24 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Ivan Kopeykin @vankop
+*/
+
+"use strict";
+
+const makeSerializable = require("./makeSerializable");
+
+class PathSet extends Set {
+	serialize({ writePathSet }) {
+		writePathSet(this);
+	}
+
+	static deserialize({ readPathSet }) {
+		const set = readPathSet();
+
+		return Object.setPrototypeOf(set, PathSet.prototype);
```

Comment:
`readPathSet` should return `PathSet `

Reply:
then `PathLazySet` must store `PathSet` instead of `Set`..

I think setting prototype is better for DX. In debugger you will see `Map`s, `Set`s instead of `PathSet`, `PathAsKeyMap`, `PathMap`. 

What do you think about this @sokra ?